lib/ZnapZend/ZFS.pm

```diff
@@ -424,9 +452,27 @@ sub sendRecvSnapshots {
     my $dstName = shift; # name of the znapzend policy => attribute prefix
     my $mbuffer = shift;
     my $mbufferSize = shift;
-    my $snapFilter = $_[0] || qr/.*/;
+    my $snapFilter = shift // qr/.*/;
+
+    # Limit creation-ordered listing after registering this snapshot name,
+    # (there may exist newer snapshots that would be not seen and replicated).
+    # For practical purposes, this can be used with --since=X mode to ensure
+    # that "X" exists on destination if it does not yet (note that if there
+    # are newer snapshots on destination, they would be removed to allow
+    # receiving "X", unless --forbidDestRollback is requested, in this case).
+    my $lastSnapshotToSee = shift // undef; # Stop creation-ordered listing after registering this snapshot name - if there is one in the filtered selection
+    if (defined($lastSnapshotToSee)) {
+        if ($lastSnapshotToSee eq "") { $lastSnapshotToSee = undef; }
+        else { $lastSnapshotToSee =~ s/^.*\@// ; }
+    }
+
+    # In certain cases, callers can set this argument to explicitly
+    # forbid (0), allow (1) or enforce if needed (2) a rollback of dest.
+    my $allowDestRollback = shift // undef;
+    if (!defined($allowDestRollback)) { $allowDestRollback = (!$self->sendRaw && !$self->forbidDestRollback) ; }
+
     my @recvOpt = $self->recvu ? qw(-u) : ();
-    push @recvOpt, '-F' if !$self->sendRaw;
+    push @recvOpt, '-F' if $allowDestRollback;
```

Comment:
is allowDestRollback connected to sendRaw ?

Reply:
I believe this partially originates from https://github.com/oetiker/znapzend/pull/496/files#diff-33cf5e498f84195d4308abccb7c59d70L355 adding the optionality of `-F` in relation to raw send (for encrypted datasets) and on another hand my PR adding `--forbidDestRollback` for people who want to explicitly not rollback existing snapshots for experiments or whatever reason.

Ultimately this PR makes it explicitly an optional argument of the `sendRecvSnapshot()` which can be set for each one sync operation in some cases (e.g. for intermediate desired snapshot in `--since` and `--sinceForced` modes) or if it is not set - defaults to the indications from global config options as they were directly used before.

Reply:
yes, if I remember correctly -F is incompatibly with sendRaw ... with this change the behavior of sendRaw will suffer, or am I missing something ?

Reply:
I hope not, but can't test. Just a few lines above this is defined as:

````
    # In certain cases, callers can set this argument to explicitly
    # forbid (0), allow (1) or enforce if needed (2) a rollback of dest.
    my $allowDestRollback = shift // undef;
    if (!defined($allowDestRollback)) { $allowDestRollback = (!$self->sendRaw && !$self->forbidDestRollback) ; }
````

so normally (if not forced via since-mode) this would allow destination rollback if we are not in sendRaw mode (as before) and if user did not pass a flag to forbidDestRollback.

Reply:
ok